### ISSUE TITLE
Google Services 4.4 on API 19 does not submit exceptions

### ISSUE THREAD

**Issue by GoogleCodeExporter** [gzog/analytics-issues#443]
```
Name of affected component: Mobile Tracking (Android)

This affects Google Analytics V4 bundled in Google Play services 4.4.52 on an 
API 19 device or emulator.  It may be isolated to just the Google Play Services 
4.4.52 but I was unable to test that on a sub API 19 device.

Issue summary:
When using ga_reportUncaughtExceptions on an 4.4.52/API 19 device exceptions 
are not dispatched. I also saw the same behavior with subclassing the GA 
Exception Handler but didn't not confirm that worked on a Non API 19 device.   
The are viewable in the log as such:

06-13 06:39:45.038: V/GAV3(4592): Thread[main,5,main]: Tracking Exception: 
NullPointerException (@AllTriviaActivity$3:onClick:75) {main}

but a log entry indicating they were dispatched never appears. However all 
other GA traffic is dispatched and viewable in reports.

If used on an API 18 emulator the exception is dispatched and is viewable on GA 
Crash/Exception report.  Running the identical code on an API 14 device also 
works.   Also downgrading the API 19 device to 4.0.34 also works. 

Steps to reproduce issue:
1. Configure Google Analytics V4 in an Android app compiling against Google 
Play Services 4.4.52 on an API 19 device/emulator with Google Play Services 
v4.4.52.

2. Use a config file for the tracker and in the manfest.  ie 
 <meta-data android:name="com.google.android.gms.analytics.globalConfigResource" android:resource="@xml/gaconfig" />

<resources xmlns:tools="http://schemas.android.com/tools"
tools:ignore="TypographyDashes">
    <string name="ga_trackingId">XXXXXXXXXXXX</string>
    <integer name="ga_sessionTimeout">30</integer>
    <bool name="ga_autoActivityTracking">true</bool>
    <bool name="ga_reportUncaughtExceptions">true</bool>
    <string name="ga_logLevel">verbose</string>
</resources>

3. Force an crash to occur and observe the log and GA Crash/Exception report.

Expected output:
The log should indicate that the exception has been caught by GA and passed on 
to the default handler.  Then the dispatch to the server should be apparent. 
The crash/exception should also show up in the report.

Actual results:
The crash is caught but never dispatched or indicated that it is saved to the 
hit store.  The exception never shows up on the GA Crash Exception report even 
after waiting several days.  Normal GA pages/event hits do appear in their 
respective reports and the realtime reports.

Notes:
This appears related to the integrated dispatching that is present in Google 
Play Services 4.4.52. As mentioned above I downgraded my API 19 device to 
4.0.34 and the exception was stored to the hit store and was later dispatched. 
The exception later showed in the reports.  I have read of other people having 
a similar problem of crashes/exceptions not showing up in reports with other 
devices.  I am unsure of their version of Google Play Services they were using 
but it was likely 4.4 due to the recent dates of the postings. 

```

Original issue reported on code.google.com by `spaceast...@gmail.com` on 13 Jun 2014 at 12:04

**Comment by GoogleCodeExporter** [comment]
```
I'm seeing the same lack of exception reporting with GPS 5.0.84 and Android 
4.4.2.
```

Original comment by `pushbit` on 7 Jul 2014 at 7:59

**Comment by GoogleCodeExporter** [comment]
```
I am seeing the same issue that exceptions are not being dispatched.
Is it accepted as a bug yet?
```

Original comment by `vvitala...@ipass.com` on 1 Aug 2014 at 1:10

**Comment by GoogleCodeExporter** [comment]
```
Can Google accept that this is a bug, so that developers can use Analytics v3 
and not migrate the code to v4
```

Original comment by `venki...@gmail.com` on 1 Aug 2014 at 1:36

**Comment by GoogleCodeExporter** [comment]
```
Are you guys sure that this issue is only present on devices with GPS 5.0.84 
and Android 4.4.2? I upgraded Analytics to v4 and I no longer see exceptions 
being dispatched on any of my devices (also running earlier versions of 
Android).
```

Original comment by `Peter.R...@inova.si` on 4 Aug 2014 at 6:30

**Comment by GoogleCodeExporter** [comment]
```
Actually it is not related to the Android OS version. 
It is related to the Google Play Services running on the device. 
If the google play services is of version 5.0.77, you will see this issue. 
Which means there may be devices on a lower Android Version but having latest 
google play services, you should see this issue. I am moving back to google 
analytics V3 and will move to Analytics V4 once we hear more on this issue.
```

Original comment by `vvitala...@ipass.com` on 4 Aug 2014 at 9:24

**Comment by GoogleCodeExporter** [comment]
```
I'm pretty sure it affects every version of Google Play Services 4.4 and above 
regardless of the API version.  I just was unable to test it for sure on a real 
device. However, when I did use it with the emulator using lower API versions 
that didn't have the latest GPS installed (4.4.52 at the time) the crashes 
dispatched properly and were recorded in the reports.  
```

Original comment by `spaceast...@gmail.com` on 5 Aug 2014 at 1:20

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
I'm using Google Play Services 5.0.89-000. I followed the steps detailed here: 
https://developers.google.com/analytics/devguides/collection/android/v4/exceptio
ns, using "Automatic Configuration".

I'm getting this output in the LogCat when an unhandled exception is thrown:

[...]
08-22 09:28:57.862: V/GAV4(28731): Thread[main,5,main]: Tracking Exception: 
NullPointerException (@MyFragment$3:onClick:311) {main}
08-22 09:28:57.862: V/GAV4(28731): Thread[main,5,main]: Dispatch call queued. 
Dispatch will run once initialization is complete.
08-22 09:28:57.870: V/GAV4(28731): Thread[main,5,main]: Passing exception to 
original handler.
[...]

...then the Applications closes due to the Exception. The next time the 
Application is restarted, this is formation is shown in LogCat:

[...]
08-22 09:34:53.636: V/GAV4(28902): Thread[main,5,main]: ExceptionReporter 
created, original handler is com.android.internal.os.RuntimeInit$UncaughtHandler
08-22 09:34:53.636: V/GAV4(28902): Thread[main,5,main]: Uncaught exceptions 
will be reported to Google Analytics.
08-22 09:34:58.612: V/GAV4(28902): Thread[GAThread,5,main]: connecting to 
Analytics service
08-22 09:34:58.612: V/GAV4(28902): Thread[GAThread,5,main]: connect: 
bindService returned true for Intent { 
act=com.google.android.gms.analytics.service.START 
cmp=com.google.android.gms/.analytics.service.AnalyticsService (has extras) }
[...]

But the Exception is never shown in the "GA Crashes and Exceptions"


I'm experiencing this behavior in these Real devices:

- HTC Desire Z with API10
- Galaxy Nexus with API15
- Galaxy SIII with API16
- Sony Xperia Z with API17
- Nexus 7 with API19
```

Original comment by `josego...@gmail.com` on 25 Aug 2014 at 7:23

**Comment by GoogleCodeExporter** [comment]
```
I got this issue with API 18 and 19 using Google Play Services 5.0.89 
(1307510-034) on real device Samsung Galaxy S3 mini (Android 4.1.2). See 
https://stackoverflow.com/questions/25351648/how-to-get-google-analytics-v4-to-r
eport-uncaught-exceptions-crashes-for-my-ap for my original post.
```

Original comment by `oakw...@gmail.com` on 29 Aug 2014 at 3:02

**Comment by GoogleCodeExporter** [comment]
```
We are seeing the same issue, and hoping this bug is going to be fixed soon.
```

Original comment by `stouton...@gmail.com` on 5 Sep 2014 at 10:16

**Comment by GoogleCodeExporter** [comment]
```
I can confirm seeing the same issue on 4.4.2 and 4.0.4 both having up-to-date 
(4.4.52) GPS, but not on another device running 2.3.4 having no GPS installed.
```

Original comment by `sfthoma...@gmail.com` on 6 Sep 2014 at 9:41

**Comment by GoogleCodeExporter** [comment]
```
Same issue here with Google Play Services 5.0.89 on real HTC One M8 with 
Android 4.4.3. I have tried both ExceptionReporter and the automatic setting. 
No exceptions show up in Crashes&Exceptions report.
```

Original comment by `mb040...@gmail.com` on 13 Sep 2014 at 8:57

**Comment by GoogleCodeExporter** [comment]
```
Not Sure if there is any use commenting here??? Have not heard of anything from 
any of Analytics Developers. We are seeing the same issue too. This was 
reported on June 13 and till now we do not have any confirmation from Google 
Analytics team that this is an issue and they would fix it OR if this is not an 
issue on their side.

Please Help Google Analytics Team.

Thanks,
Venki
```

Original comment by `venki...@gmail.com` on 15 Sep 2014 at 4:45

**Comment by GoogleCodeExporter** [comment]
```
The development team probably wouldn't care for issues with few stars.  So far 
this issue has got only 24 stars...
```

Original comment by `stouton...@gmail.com` on 17 Sep 2014 at 5:29

**Comment by GoogleCodeExporter** [comment]
```
Thankfully I found this thread. I kept thinking I had missed out something. 
Hope this bug is fixed soon.
```

Original comment by `lakshman...@gmail.com` on 23 Sep 2014 at 12:18

**Comment by GoogleCodeExporter** [comment]
```
Any comment from the development team? I hope to know at least if it's a bug 
and is going to be fixed or not.
```

Original comment by `stouton...@gmail.com` on 29 Sep 2014 at 4:02

**Comment by GoogleCodeExporter** [comment]
```
The team is aware and working on a fix.
```

Original comment by `pfrise...@google.com` on 30 Sep 2014 at 3:35
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Some progress??
```

Original comment by `ern...@timo.com.br` on 16 Oct 2014 at 6:49

**Comment by GoogleCodeExporter** [comment]
```
What is the status of this issue?  Is it in progress or still waiting to be 
started?
```

Original comment by `stouton...@gmail.com` on 31 Oct 2014 at 8:02

**Comment by GoogleCodeExporter** [comment]
```
it's in progress. 
```

Original comment by `pfrise...@google.com` on 31 Oct 2014 at 4:15
- Changed state: **Prioritized**

**Comment by GoogleCodeExporter** [comment]
```
i have the same issue
is there any bug fix yet?
thanks!
```

Original comment by `ja.moham...@gmail.com` on 20 Dec 2014 at 6:03

**Comment by GoogleCodeExporter** [comment]
```
So many crashes have been hidden to developpers since june ....
```

Original comment by `mad...@gmail.com` on 26 Dec 2014 at 5:22

**Comment by GoogleCodeExporter** [comment]
```
when to fix it??
```

Original comment by `clun...@gmail.com` on 7 Jan 2015 at 7:22

**Comment by GoogleCodeExporter** [comment]
```
There were some improvements in the last release but depending on when/where 
the app crashes it may not always be possible to dispatch hits. There are 
additional improvements that will be in the next release of Google Play 
services to help with this issue.

If you've noticed any improvements from the last release please let us know. If 
you're still experiencing this issue, some additional details (trace) would be 
helpful.
```

Original comment by `pfrise...@google.com` on 9 Jan 2015 at 8:08

**Comment by GoogleCodeExporter** [comment]
```
when will I see this "next" release of play services
```

Original comment by `mad...@gmail.com` on 2 Feb 2015 at 12:26

**Comment by GoogleCodeExporter** [comment]
```
Hi I have tried google play service rev 21,still this issue is present.I have 
teste on sony zperia p and google nexus 7
```

Original comment by `amit.des...@gmail.com` on 6 Mar 2015 at 8:00

**Comment by GoogleCodeExporter** [comment]
```
It seems to be working at least part of the time now.  However, those 
exceptions I am seeing don't seem to have any model or os version attached in 
the reports.
```

Original comment by `spaceast...@gmail.com` on 18 Mar 2015 at 10:51

**Comment by GoogleCodeExporter** [comment]
```
Hi , I tried google play services rev.23 on API 19 ,but this issue hasn't 
fixed.  
I'm waiting for fixing...
```

Original comment by `bucho.hi...@gmail.com` on 23 Mar 2015 at 4:15

**Comment by GoogleCodeExporter** [comment]
```
google team, what is happening with you??? Why should developers look for 
workarounds to get that analytics done fo 6 months? Still can't get that 
working...
```

Original comment by `nerses.z...@gmail.com` on 7 Apr 2015 at 3:36

**Comment by GoogleCodeExporter** [comment]
```
Dear Google, this bug is a blocker for a ticket on our roadmap since September 
2014. Luckily we had a lot of other things to do but this one is getting more 
and more essential since the biggest part of functionality has already been 
implemented and we are now dedicated our resources to app's stability and 
performance issues. 
We don't want to implement some sneaky solutions nor we want to use the 
previous GA versions. Please tell us if the bug will be fixed and when it is 
going to happen. 
```

Original comment by `N.Tiho...@gmail.com` on 8 Apr 2015 at 7:34

**Comment by GoogleCodeExporter** [comment]
```
Still not fixed for Lollipop 5.1 with google play service 7.
```

Original comment by `h...@haystack.tv` on 16 Apr 2015 at 9:37

**Comment by GoogleCodeExporter** [comment]
```
See Release Version 4.5 - Google Play Services 7.3 (May 1, 2015)
https://developers.google.com/analytics/devguides/collection/android/changelog
```

Original comment by `pfrise...@google.com` on 5 May 2015 at 6:52
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
It seems that this issue got fixed now. I have updated to new version of Google 
Play Services and done initial checks. Uncaught exceptions are reported. Yet to 
complete the final testing... but thanks google... we were affected lot because 
of this issue.
```

Original comment by `lpravink...@gmail.com` on 6 May 2015 at 7:50

**Comment by GoogleCodeExporter** [comment]
```
With play-services:7.5.0 it's working.
```

Original comment by `rodol...@moofwd.com` on 19 Aug 2015 at 3:42

**Comment by GoogleCodeExporter** [comment]
```
Can this be closed? GPS 7.8.0 is out now.
```

Original comment by `jaredsburrows@gmail.com` on 19 Aug 2015 at 4:20

**Comment by GoogleCodeExporter** [comment]
```
I am using 1.3

I can see that the call to trackPageview is incrementing the retain count for 
the pageURL parameter by 2 http://ccl6t.com/
```

Original comment by `musicnot...@gmail.com` on 11 Nov 2015 at 8:56
